lib/session.go

```diff
@@ -23,22 +23,50 @@ type Session struct {
 	SSHKeys    map[string]string `json:"ssh_keys,omitempty"`
 }
 
-func (e *Session) Assume(arn string) (*Session, error) {
+func (e *Session) Assume(providedArn string) (*Session, error) {
 	expiration := e.Expiration
 	maxExpiration := time.Now().Add(time.Hour).Truncate(time.Second)
 	if expiration.After(maxExpiration) {
 		expiration = maxExpiration
 	}
 
 	duration := expiration.Sub(time.Now())
-	creds, err := e.AWSCreds.AssumeRole(arn, duration)
-	if err != nil {
-		return nil, err
+
+	var creds *AWSCredentials
+
+	selectedArn := providedArn
+	arnStruct, err := ParseARN(providedArn)
+	if err == nil {
+		creds, err = e.AWSCreds.AssumeRole(arnStruct.String(), duration)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Unparseable ARN
+		identityArn, err := e.AWSCreds.GetCallerIdentity()
+		if err != nil {
+			return nil, err
+		}
+
+		roleArn := &ARN{
+			Partition: identityArn.Partition,
+			Service:   "iam",
+			Region:    identityArn.Region,
```

Comment:
😒 Sorry I didn't catch this before… but there isn't a `Region` component specified for a role ARN (IAM roles are global). See [Amazon Resource Names (ARNs) and AWS Service Namespaces](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arn-syntax-iam)

Reply:
(read: we should set this to `""`)